### ISSUE TITLE
[RB] About RenameInstanceVariableDriver logic

### ISSUE THREAD

**Issue by Ducasse** [guillep/pharo-core#15780]
Hi Caro and Balsa

Here are some notes about the following method. 

```
prepareRefactoringInteractively
	| failedConditions |
	
	"We ask the user for the variable new name"
	[ newName := self requestNewNameBasedOn: lastName.
	shouldEscape ifTrue: [ ^ self ].
	refactoring newName: newName.
	failedConditions := refactoring failedApplicabilityPreconditions.
	failedConditions isEmpty ]
		whileFalse: [
			failedConditions do: [ :cond |
				self application newInform
					label: cond errorString;
					title: 'Invalid name';
					openModal ] ].
```

I realized that I do not like too much the methods above. 
- I do not think that we want to use driver in non interactive mode. Do we have a scenario for this?
So this could simplify the runRefactoring and a part of the logic of this method could be pushed into runRefactoring. 
I like the flow of the RemoveClassRefactoring for example. 
- I do not like the idea behind a list of failed Conditions especially because we loop over them so it would mean that the user could have a flood of pop ups. 
So tell me.

**Comment by Ducasse** [comment]
@balsa-sarenac 

May be Caro already worked on this in the pending PR. 
I was just looking at it in the context of my work on RBRenameMethod.

**Comment by balsa-sarenac** [comment]
Sorry I missed this issue earlier. I agree with all your points, I had the same concerns myself. If it makes sense, I can spend some time working on this, and then get back to extract method?

**Comment by Ducasse** [comment]
Thanks Balsa. Let us wait for the PR of Caro and then it would be good that you do a pass. 
I'm experiencing some pain with RBRenameMethod too. In particular the mix of applicabilityPreconditions and preconditions.
(I learned that I want a show the sender of message x in the hierarchy above me).

**Comment by Ducasse** [comment]
@balsa-sarenac I push my current effort into https://github.com/Ducasse/pharo/tree/enh/RenameMethodDriverUpdate
if you want to have a look.

<blockquote><img src="https://opengraph.githubassets.com/a3d6502b0f1d3450d1917acd0d902257ef3f7dfcf160558fccea97dd56b56389/Ducasse/pharo" width="48" align="right"><div><img src="https://github.githubassets.com/favicons/favicon.svg" height="14"> GitHub</div><div><strong><a href="https://github.com/Ducasse/pharo">GitHub - Ducasse/pharo at enh/RenameMethodDriverUpdate</a></strong></div><div>Pharo is a dynamic reflective pure object-oriented language supporting live programming inspired by Smalltalk. - GitHub - Ducasse/pharo at enh/RenameMethodDriverUpdate</div></blockquote>

**Comment by balsa-sarenac** [comment]
Yes, I will take a look. If it makes sense we can touch base tomorrow or Thursday, I am available.

**Comment by Ducasse** [comment]
Ok I will let you know because I have visitors around

**Comment by carolahp** [comment]
Hello @Ducasse and @balsa-sarenac , sorry I just saw this issue. 
I agree with Stef, the architecture does not fit the one we are using as template. 
The need for "non interactive" mode arises from tests. 

We can discuss later today if you are available. 
I propose finishing the work on [PR](https://github.com/pharo-project/pharo/pull/15774) and then I can take care on fixing the architecture of both rename instance and shared variable

**Comment by Ducasse** [comment]
Sure. I was too busy today. :(

**Comment by Ducasse** [comment]
I checked and the current behavior is different and it works.
